xbmc/filesystem/NFSDirectory.cpp

```diff
@@ -200,112 +225,71 @@ bool CNFSDirectory::ResolveSymlink( const std::string &dirName, struct nfsdirent
 bool CNFSDirectory::GetDirectory(const CURL& url, CFileItemList &items)
 {
   // We accept nfs://server/path[/file]]]]
-  int ret = 0;
-  KODI::TIME::FileTime fileTime, localTime;
   std::unique_lock lock(gNfsConnection);
-  std::string strDirName="";
-  std::string myStrPath(url.Get());
-  URIUtils::AddSlashAtEnd(myStrPath); //be sure the dir ends with a slash
 
-  if(!gNfsConnection.Connect(url,strDirName))
+  std::string strDirName = "";
+  if (!gNfsConnection.Connect(url, strDirName))
   {
     //connect has failed - so try to get the exported filesystems if no path is given to the url
-    if(url.GetShareName().empty())
-    {
-      if(url.GetHostName().empty())
-      {
-        return GetServerList(items);
-      }
-      else
-      {
-        return GetDirectoryFromExportList(myStrPath, items);
-      }
-    }
-    else
+    if (url.GetShareName().empty())
     {
-      return false;
+      std::vector<CFileItemPtr> fileItems =
+          url.GetHostName().empty() ? GetServerList() : GetDirectoryFromExportList(url);
+      bool ret = !fileItems.empty();
+      items.AddItems(std::move(fileItems));
+      return ret;
     }
+    return false;
   }
 
-  struct nfsdir *nfsdir = NULL;
-  struct nfsdirent *nfsdirent = NULL;
-
-  ret = nfs_opendir(gNfsConnection.GetNfsContext(), strDirName.c_str(), &nfsdir);
-
-  if(ret != 0)
+  struct nfsdir* nfsdir = nullptr;
+  if (nfs_opendir(gNfsConnection.GetNfsContext(), strDirName.c_str(), &nfsdir) != 0)
   {
     CLog::Log(LOGERROR, "Failed to open({}) {}", strDirName,
               nfs_get_error(gNfsConnection.GetNfsContext()));
     return false;
   }
   lock.unlock();
 
-  while((nfsdirent = nfs_readdir(gNfsConnection.GetNfsContext(), nfsdir)) != NULL)
+  std::string myStrPath(url.Get());
+  URIUtils::AddSlashAtEnd(myStrPath);
+  URIUtils::AddSlashAtEnd(strDirName);
+
+  std::string resolvedPath;
+  std::vector<CFileItemPtr> fileItems;
+  struct nfsdirent* dirent = nullptr;
+  while ((dirent = nfs_readdir(gNfsConnection.GetNfsContext(), nfsdir)) != nullptr)
   {
-    struct nfsdirent tmpDirent = *nfsdirent;
-    std::string strName = tmpDirent.name;
-    std::string path(myStrPath + strName);
-    int64_t iSize = 0;
-    bool bIsDir = false;
-    int64_t lTimeDate = 0;
+    const std::string& name = dirent->name;
 
     //resolve symlinks
-    if(tmpDirent.type == NF3LNK)
-    {
-      CURL linkUrl;
-      //resolve symlink changes tmpDirent and strName
-      if(!ResolveSymlink(strDirName,&tmpDirent,linkUrl))
-      {
-        continue;
-      }
-
-      path = linkUrl.Get();
-    }
+    //resolve symlink changes dirent and name
+    const bool isSymLink = dirent->type == NF3LNK;
+    if (isSymLink && !ResolveSymlink(strDirName, dirent, resolvedPath))
+      continue;
 
-    iSize = tmpDirent.size;
-    bIsDir = tmpDirent.type == NF3DIR;
-    lTimeDate = tmpDirent.mtime.tv_sec;
+    if (name == "." || name == ".." || name == "lost + found")
```

Comment:
Before we checked for `lost+found` now it's `lost + found` is that intentional?

Reply:
This is copy/paste typo due my code in my previous comment :(